### PR TITLE
Replace Cancel with Terminate

### PR DESCRIPTION
**Pull request by hustclf**
- replace Cancel with Terminate

- optimize imports

### CODE REVIEW

**Comment by hustclf** [comment]
Please review it again :)  @mrooding

**Comment by hustclf** [comment]
Anyone who can help to review the code for the pr ? thanks.

@mrooding @Nick-Triller @nielsdenissen

File: cmd/cli/flink/terminate.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	if len(mode) > 0 {
+		url = c.constructURL(fmt.Sprintf("jobs/%v?mode=%v", jobID, mode))
+	} else {
+		url = c.constructURL(fmt.Sprintf("jobs/%v", jobID))
+	}
+
+	req, err := c.newRequest("PATCH", url, nil)
+
+	// set retry policy: do not retry when status code is 500. (indicating the job is not stoppable)
+	c.Client.CheckRetry = func(ctx context.Context, resp *http.Response, err error) (b bool, e error) {
```
Comment (mrooding) [inline cmd/cli/flink/terminate.go]: I'd prefer to split this off into a separate function in this file

File: cmd/cli/main.go [hunk 2]
```diff
@@ -168,6 +168,34 @@ func UpdateAction(c *cli.Context) error {
...
+		return cli.NewExitError("unspecified flag 'job-name-base'", -1)
+	}
+
+	mode := c.String("mode")
+	if len(mode) > 0 && mode != "cancel" && mode != "stop" {
+		return cli.NewExitError("unknown value for 'mode', only 'cancel' and 'stop' are supported", -1)
+	}
+
+	terminate.Mode = mode
+	err := operator.Terminate(terminate)
```
Comment (mrooding) [inline cmd/cli/main.go]: Nitpicking but empty line before 188 and remove the empty line on 189

File: cmd/cli/flink/terminate.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+	ErrInfo string `json:"error"`
+}
+
+// Terminate terminates a running job specified by job ID
+func (c FlinkRestClient) Terminate(jobID string, mode string) error {
+	var url string
+	if len(mode) > 0 {
+		url = c.constructURL(fmt.Sprintf("jobs/%v?mode=%v", jobID, mode))
+	} else {
+		url = c.constructURL(fmt.Sprintf("jobs/%v", jobID))
```
[the remark this answers is not in the document]
Comment (hustclf) [reply cmd/cli/flink/terminate.go]: yes, you are right, I will optimize it.

File: cmd/cli/main.go (hunk 2 above)
Comment (hustclf) [reply cmd/cli/main.go]: I agree, there is no need for the empty line.

**Comment by hustclf** [comment]
All the changes is  resolved, please check it again.

**Comment by hustclf** [comment]
All the changes is  resolved, please check it again.

**Comment by hustclf** [comment]
Document about `terminate` is also added,  I am grateful if you could take a look again :) @mrooding

File: cmd/cli/main.go [hunk 4]
```diff
@@ -168,6 +168,33 @@ func UpdateAction(c *cli.Context) error {
 	return nil
 }
 
+// TerminateAction executes the CLI terminate command
+func TerminateAction(c *cli.Context) error {
+	terminate := operations.TerminateJob{}
+
+	jobNameBase := c.String("job-name-base")
+	if len(jobNameBase) != 0 {
```
Comment (mrooding) [inline cmd/cli/main.go]: With the mode you only have an if clause and after the if you assing `terminate.Mode = mode`. May be neater to do the same with this argument:

```
jobNameBase := c.String("job-name-base")
if len(jobNameBase) == 0 {
  return cli.NewExitError("unspecified flag 'job-name-base'", -1)
}
terminate.JobNameBase = jobNameBase
```

File: cmd/cli/main.go [hunk 5]
```diff
@@ -168,6 +168,33 @@ func UpdateAction(c *cli.Context) error {
...
+		terminate.JobNameBase = jobNameBase
+	} else {
+		return cli.NewExitError("unspecified flag 'job-name-base'", -1)
+	}
+
+	mode := c.String("mode")
+	if len(mode) > 0 && mode != "cancel" && mode != "stop" {
+		return cli.NewExitError("unknown value for 'mode', only 'cancel' and 'stop' are supported", -1)
+	}
```
Comment (mrooding) [inline cmd/cli/main.go]: Nitpicking again but put `terminate.Mode = mode` directly on 186 directly after the closing bracket on 185 and keep 1 empty line between 186 and the current 188 where you call `operator.Terminate`

**Comment by mrooding** [comment]
Thank you very much for your contribution and hard work @hustclf

**Pull request merged**